src/OmniSharp.Roslyn/OmniSharpWorkspace.cs

```diff
@@ -156,6 +157,15 @@ public void TryPromoteMiscellaneousDocumentsToProject(Project project)
             }
         }
 
+        public void UpdateRulesetsForProject(ProjectId projectId, ImmutableDictionary<string, ReportDiagnostic> newRules)
+        {
+            var project = this.CurrentSolution.GetProject(projectId);
+            var existingRules = project.CompilationOptions.SpecificDiagnosticOptions;
+            var distinctRulesWithProjectSpecificRules = newRules.Concat(existingRules.Where(x => !newRules.Keys.Contains(x.Key)));
```

Comment:
>var distinctRulesWithProjectSpecificRules = newRules.Concat(existingRules.Where(x => !newRules.Keys.Contains(x.Key))); [](start = 12, length = 118)

Given that this adds both the new rules and the old rules that aren't in the new ones, how does someone *remove* a rule?

Reply:
Valid point: i think i need to find clean version of `SpecificDiagnosticOptions` and use that one as base instead of project own possibly already updated version.

Reply:
Fixed by https://github.com/OmniSharp/omnisharp-roslyn/pull/1440/commits/a7aa1fd617cee604073eb835b6328c1ce0a1f858